src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -419,6 +426,7 @@ void DefaultOTARequestor::OnConnected(void * context, Messaging::ExchangeManager
 {
     DefaultOTARequestor * requestorCore = static_cast<DefaultOTARequestor *>(context);
     VerifyOrDie(requestorCore != nullptr);
+    requestorCore->mSessionHolder.Grab(sessionHandle);
```

Comment:
Should we be clearing this holder in `OnConnectionFailure` in case through some sequence of calls, there's still a handle in there from a previous successful connection attempt?